pallets/automation-time/src/lib.rs

```diff
@@ -64,21 +64,32 @@ use frame_support::{
 	weights::constants::WEIGHT_REF_TIME_PER_SECOND,
 };
 use frame_system::pallet_prelude::*;
+use hex;
 use orml_traits::{FixedConversionRateProvider, MultiCurrency};
 use pallet_parachain_staking::DelegatorActions;
 use pallet_timestamp::{self as timestamp};
 pub use pallet_xcmp_handler::InstructionSequence;
 use pallet_xcmp_handler::XcmpTransactor;
 use primitives::EnsureProxy;
-use scale_info::TypeInfo;
+use scale_info::{prelude::format, TypeInfo};
 use sp_runtime::{
-	traits::{CheckedConversion, Convert, Dispatchable, SaturatedConversion, Saturating},
+	traits::{
+		CheckedConversion, CheckedSub, Convert, Dispatchable, SaturatedConversion, Saturating,
+	},
 	ArithmeticError, DispatchError, Perbill,
 };
 use sp_std::{boxed::Box, vec, vec::Vec};
 pub use weights::WeightInfo;
 use xcm::{latest::prelude::*, VersionedMultiLocation};
 
+#[macro_export]
+macro_rules! ERROR_MESSAGE_DELEGATION_FORMAT {
```

Comment:
In Rust, the format! macro requires the formatting string to be a string literal. We cannot use constants or static variables directly as the formatting string. To address this, we can use a macro to extract the formatting string.

Reply:
I don’t think we need an extra `hex` library to encode the wallets. Can you write down the actual message of the `error_message` here? Is the wallet address still human readable after hex encoding?

On second thought, if we were to add more details, we should always print out delegator and collator for all AutoCompoundDelegatorStake events in a way that is easier to parse.

For example, 
AutoCompoundDelegatorStakeSucceeded
- taskId
- amount
- collator

AutoCompoundDelegatorStakeSkipped
- taskId
- amount
- collator
- message: `Unable to pay fees for the next execution`
- value: <the expected amount to pay for the next execution>
> The balance reason needs an explanation, as in what was the balance of the wallet, so we add that information by adding an extra field to this event. The challenge is to find a common method to add extra information to a message, or an error. Let me ask round to find out the best practice to extend the Error definition. @v9n any recommendations? We need to standardize the format of Success and Error events.

AutoCompoundDelegatorStakeFailed
- taskId
- amount
- collator
- error: DelegationNotFound
> Since there’s already a collator information included, we don’t need any extra fields for this `AutoCompoundDelegatorStakeFailed` error

Reply:
After undergoing `hex` processing, it will become an account ID string.
For example: 0x3830ef2fdb02d711afe91c830737375e32ac9904886266897f361b5a2ec13a39

Reply:
Yes, wallet addresses need to be printed out, but not hex or accountId. @imstar15